src/update_feed.py

```diff
@@ -141,114 +169,298 @@ def _parse_link_header(headers: dict) -> dict[str, str] | None:
+                    if self.is_fork and " (" in malware_name:
+                        malware_name = malware_name[: malware_name.index(" (")]
+                    malware_name += f" ({fork.owner}/{fork.name})"
+
+                    yield GithubRepo(
+                        url=fork.url,
+                        category=self.category,
+                        malware_name=malware_name,
+                        name=fork.name,
+                        owner=fork.owner,
+                    )
+        except QuotaLimitReached:
+            yield None  # indicate quota limit was reached
 
     def get_asset_hashes_from_release(
-        self, release: dict
+        self, release: Release
     ) -> tuple[SemVer, list[tuple[str, str]]]:
-        tag_name = release.get("tag_name", None)
-        version = normalize_version(tag_name)
-        variant_pfx = None
-        if version == SemVer(0, 0, 0):
-            # if the tag is not a proper version, use it as variant prefix instead
-            # and use the date of the release as the version instead
-            variant_pfx = tag_name
-            published_at_str = release.get("published_at", "")
-            published_at = datetime.datetime.fromisoformat(published_at_str)
-            version = SemVer(
-                published_at.year,
-                published_at.month,
-                published_at.day,
-                label=f"{published_at.hour:02}:{published_at.minute:02}",
-            )
-
-        release_name = release.get("name", None)
-        if release_name is None or release_name == "'":
-            logger.info("No release name provided in release")
+        variant_pfx = release.tag if release.tag_as_variant else None
 
-        assets = map(self._parse_asset, release.get("assets", []))
-        variant_hashes = process_assets(
+        assets = map(self._process_asset, release.asset_entries)
+        variant_hashes = calculate_asset_hashes(
             assets,
             self.name,
             self.malware_name,
             release,
             variant_prefix=variant_pfx,
         )
-        return version, list(variant_hashes)
+        return release.version, list(variant_hashes)
 
     @staticmethod
-    def _parse_asset(asset: dict) -> Asset:
+    def _process_asset(asset: dict) -> Asset:
         return Asset(
             name=asset.get("name", ""),
             dl_url=asset["browser_download_url"],
             created_at=asset.get("created_at", ""),
             content_type=asset.get("content_type", ""),
         )
 
+    @staticmethod
+    def _process_release(release: dict) -> Release:
+        tag_name = release.get("tag_name", "")
+        version = normalize_version(tag_name)
+        tag_as_variant = False  # flag to explicitely use tag in the feeds variant field
+        date = datetime.datetime.fromisoformat(release.get("published_at", ""))
+        # if no proper version can be inferred, use the release date as version
+        if version == SemVer():
```

Comment:
What's the benefit of `normalize_version` returning not None in case of failure? Does `SemVer` have `__eq__` defined so that this comparison works? If not, maybe just returning None would be more reliable if new fields are added and `__eq__` is not updated to account for them.

Reply:
Yes,`__eq__` is created by default in a dataclass. In addition, `SemVer` has `ordered=True` enable, to also generate the corresponding ordered comparison functions per field. This way, when comparing 2 versions, it will compare: a.major > b.major, a.minor > b.minor, ... in that order. This way, there is no need to manually account for changing fields. 
The normalisation function should always return a SemVer, to avoid later type issues. What I could do instead, is return 2 values from normalization, akin to Go's `ok`-pattern. But IMHO, it adds little value here

Reply:
Thanks for the explanation. Now I understand that the current version is as good as it gets 😄